lib/msal-browser/src/cache/TokenCache.ts

```diff
@@ -81,149 +80,76 @@ export class TokenCache implements ITokenCache {
         response: ExternalTokenResponse,
         options: LoadTokenOptions
     ): AuthenticationResult {
-        this.logger.info("TokenCache - loadExternalTokens called");
-
-        if (!response.id_token) {
+        if (!this.isBrowserEnvironment) {
             throw createBrowserAuthError(
-                BrowserAuthErrorCodes.unableToLoadToken
+                BrowserAuthErrorCodes.nonBrowserEnvironment
             );
         }
 
-        const idTokenClaims = AuthToken.extractTokenClaims(
-            response.id_token,
-            base64Decode
+        const idTokenClaims = response.id_token
+            ? AuthToken.extractTokenClaims(response.id_token, base64Decode)
+            : undefined;
+
+        const authorityOptions: AuthorityOptions = {
+            protocolMode: this.config.auth.protocolMode,
+            knownAuthorities: this.config.auth.knownAuthorities,
+            cloudDiscoveryMetadata: this.config.auth.cloudDiscoveryMetadata,
+            authorityMetadata: this.config.auth.authorityMetadata,
+            skipAuthorityMetadataCache:
+                this.config.auth.skipAuthorityMetadataCache,
+        };
+        const authority = request.authority
+            ? new Authority(
+                  Authority.generateAuthority(
+                      request.authority,
+                      request.azureCloudOptions
+                  ),
+                  this.config.system.networkClient,
+                  this.storage,
+                  authorityOptions,
+                  this.logger,
+                  request.correlationId || BrowserCrypto.createNewGuid()
+              )
+            : undefined;
+
+        const cacheRecordAccount: AccountEntity = this.loadAccount(
+            request,
+            options.clientInfo || response.client_info || "",
+            idTokenClaims,
+            authority
```

Comment:
Technically `authority` can be undefined here and indirectly fails inside `loadAccount()`. Is that intentional?